swap/swap.go

```diff
@@ -346,6 +349,26 @@ func (s *Swap) sigHashCheque(cheque *Cheque) []byte {
 	return crypto.Keccak256([]byte(withPrefix))
 }
 
+// verifyChequeSig verifies the signature on the cheque
+func (s *Swap) verifyChequeSig(cheque *Cheque, expectedSigner common.Address) error {
+	sigHash := s.sigHashCheque(cheque)
+
+	sig := make([]byte, len(cheque.Sig))
+	copy(sig, cheque.Sig)
+	sig[len(sig)-1] -= 27
+	pubKey, err := crypto.SigToPub(sigHash, sig)
+
+	if err != nil {
+		return err
+	}
+
+	if crypto.PubkeyToAddress(*pubKey) != expectedSigner {
+		return ErrInvalidChequeSignature
+	}
+
+	return err
```

Comment:
At this point no error can possibly be returned (right?), so it is semantically clearer to return `return nil`